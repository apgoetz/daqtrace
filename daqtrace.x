SECTIONS
{
  /* `0` specifies the start address of this virtual (`(INFO)`) section */
  .daqtrace 0 (INFO) :
  {

    /* Everything user-defined */
    *(.daqtrace);
  }
}

ASSERT(SIZEOF(.daqtrace) < (1 << 16), ".daqtrace section id must fit in u16");
