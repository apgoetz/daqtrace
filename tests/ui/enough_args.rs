use daqtrace::trace;
mod common;


// trace macro requires two parameters
fn main() {
    let sink = common::DummySink;
    trace!(sink);
}

