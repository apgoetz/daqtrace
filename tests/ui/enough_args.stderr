error: trace! macro requires at least 2 parameters
 --> $DIR/enough_args.rs:8:5
  |
8 |     trace!(sink);
  |     ^^^^^^^^^^^^^
  |
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
